corehq/apps/locations/fixtures.py

```diff
@@ -229,57 +230,83 @@ def should_sync_flat_fixture(project, app):
+          <related_location distance="X">location_b</related_location>
+        </location>
+        <location id="location_b">
+          <related_location distance="X">location_a</related_location>
+        </location>
+      </locations>
+    </fixture>
     """
     id = 'related_locations'
-    serializer = RelatedLocationSerializer()
 
     def __call__(self, restore_state):
         if not toggles.RELATED_LOCATIONS.enabled(restore_state.domain):
             return []
 
-        restore_user = restore_state.restore_user
-        user_locations = restore_user.get_sql_locations(restore_user.domain)
-        related_location_ids = LocationRelation.from_locations(user_locations)
-        related_location_pks = (
-            SQLLocation.objects.filter(location_id__in=related_location_ids)
-            .values_list('pk', flat=True)
+        location_relations = self._users_related_locations(restore_state.restore_user)
+
+        root_node = Element('fixture', {'id': self.id,
+                                        'user_id': restore_state.restore_user.user_id,
+                                        'indexed': 'true'})
+        outer_node = SubElement(root_node, 'locations')
+
+        for location, relations in location_relations:
+            location_node = SubElement(outer_node, 'location', {'id': location.location_id})
+            for location, distance in relations:
+                node = SubElement(location_node, 'related_location')
+                node.text = location.location_id
+                if distance:
+                    node.attrib['distance'] = str(distance)
+
+        return [get_index_schema_node(self.id, ['@id']), root_node]
+
+    def _users_related_locations(self, restore_user):
+        """Returns a sorted list of location relations:
+            [
+                (location_a, [(location_b, distance), ...]),
+                (location_b, [(location_a, distance), ...])
+            ]
+
+        Sorted by the location's name, and each associated list is sorted by its location names.
+        This is purely for deterministicly ordered outputs, and can be changed if needed.
+        """
+        user_location_ids = restore_user.get_location_ids(restore_user.domain)
+        user_locations_with_descendants = SQLLocation.objects.get_descendants(
+            Q(domain=restore_user.domain, location_id__in=user_location_ids)
         )
-        locations_queryset = _location_queryset_helper(restore_user.domain, list(related_location_pks))
-        data_fields = _get_location_data_fields(restore_user.domain)
-        return self.serializer.get_xml_nodes(self.id, restore_user, locations_queryset, data_fields)
+        location_relations = LocationRelation.objects.filter(
+            Q(location_a__in=user_locations_with_descendants) | Q(location_b__in=user_locations_with_descendants)
+        ).prefetch_related('location_a', 'location_a__location_type', 'location_b', 'location_b__location_type')
```

Comment:
Its likely this can be done in fewer (perhaps custom) queries. I consciously made the decision to keep it like this to make it easy to modify if necessary (there may be a workflow that is introduced where we need to also get ancestors). The performance should be sufficient from a pilot perspective